### PR TITLE
Create metadata.yaml for OpenTSDB Exporter

### PR DESCRIPTION
**Pull request by shyamvalsan**
Creating the first exporter metadata (will add the rest if this looks good)

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: exporting/opentsdb/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+overview:
+  exporter_description: |
+	Use the OpenTSDB connector for the exporting engine to archive your Netdata metrics to OpenTSDB databases for long-term storage, further analysis, or correlation with data from other sources.
+  exporter_limitations: ''
+setup:
+  prerequisites:
+    list:
+      - title: ''
+        description: |
+			- OpenTSDB and Netdata, installed, configured and operational
```
Comment (sashwathn) [inline exporting/opentsdb/metadata.yaml:21]: The indentation looks off..

File: exporting/opentsdb/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          required: false		  
+        - name: 'timeout ms'
+          default_value: '2 * update_every * 1000'
+          description: 'The timeout in milliseconds to wait for the external database server to process the data.'
+          required: false	
+        - name: 'send hosts matching'
+          default_value: 'localhost *'
+          description: 'Includes one or more space separated patterns, using * as wildcard (any number of times within each pattern). The patterns are checked against the hostname (the localhost is always checked as localhost), allowing us to filter which hosts will be sent to the external database when this Netdata is a central Netdata aggregating multiple hosts.'
+          required: false	
+		  detailed_description: |
```
Comment (sashwathn) [inline exporting/opentsdb/metadata.yaml:97]: Indentation is off

**Review by sashwathn — APPROVED** [review]

**Review by sashwathn — DISMISSED** [review]

File: exporting/opentsdb/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
+id: 'export-opentsdb'
+meta:
+  name: 'OpenTSDB'
+  link: 'https://github.com/OpenTSDB/opentsdb'
+  categories:
```
Comment (thiagoftsm) [inline exporting/opentsdb/metadata.yaml:5]: @shyamvalsan , please, take a look in `yamllint` warnings

File: exporting/opentsdb/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	- export
+  icon_filename: 'opentsdb.png'
+keywords: 
+	- exporter
+	- OpenTSDB
+	- scalable time series
+overview:
+  exporter_description: |
+	Use the OpenTSDB connector for the exporting engine to archive your Netdata metrics to OpenTSDB databases for long-term storage, further analysis, or correlation with data from other sources.
+  exporter_limitations: ''
```
Comment (thiagoftsm) [inline exporting/opentsdb/metadata.yaml:15]: @shyamvalsan ,

Add on top of this file the following lines:

```sh
# yamllint disable rule:line-length
---
```

File: exporting/opentsdb/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        title: 'Config options'
+        enabled: true
+      list:
+        - name: 'enabled'
+          default_value: 'no'
+          description: 'Enables or disables an exporting connector instance (yes|no).'
+          required: true
+        - name: 'destination'
+          default_value: 'no'
+          description: 'Accepts a space separated list of hostnames, IPs (IPv4 and IPv6) and ports to connect to. Netdata will use the first available to send the metrics..'
```
Comment (thiagoftsm) [inline exporting/opentsdb/metadata.yaml:37]: We have a typo error here (two dots at the end).

File: exporting/opentsdb/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+          required: true
+          detailed_description: |
+	     The format of each item in this list, is: [PROTOCOL:]IP[:PORT].
+
+	     PROTOCOL can be udp or tcp. tcp is the default and only supported by the current exporting engine.
+
+	     IP can be XX.XX.XX.XX (IPv4), or [XX:XX...XX:XX] (IPv6). For IPv6 you can to enclose the IP in [] to separate it from the port.
+	
+	     PORT can be a number of a service name. If omitted, the default port for the exporting connector will be used (graphite = 2003, opentsdb = 4242).
```
Comment (thiagoftsm) [inline exporting/opentsdb/metadata.yaml:46]: Why are you citing graphite port n opentsdb yaml file? :thinking: I think we can remove it.

File: exporting/opentsdb/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+            available. If the server fails to receive the data after that many failures, data loss on the connector instance is expected
+            (Netdata will also log it).
+          required: false
+        - name: 'timeout ms'
+          default_value: '2 * update_every * 1000'
+          description: 'The timeout in milliseconds to wait for the external database server to process the data.'
+          required: false
+        - name: 'send hosts matching'
+          default_value: 'localhost *'
+          description: |
```
Comment (sashwathn) [inline exporting/opentsdb/metadata.yaml:97]: I am concerned about the length of the descriptions that may make the table look ugly. But I guess we don't have much choice. :)

**Review by sashwathn — APPROVED** [review]

File: exporting/opentsdb/metadata.yaml (hunk 7 above)
Comment (ilyam8) [reply exporting/opentsdb/metadata.yaml:97]: An alternative is a simple description (that answers the question "What is it?" - e.g. "Hosts filter. The syntax is [simple patterns](https://github.com/netdata/netdata/tree/master/libnetdata/simple_pattern#simple-patterns).") and details in detailed_description. I think you tend to add details to the non-detailed description to "description".

Comment (ilyam8) [reply exporting/opentsdb/metadata.yaml:97]: An example of using this option should be added to the "Examples config" list:

```yaml
- name: Using `send hosts matching`
  description: ...
  config: ...
- name: Using `send charts matching`
  description: ...
  config: ...

Comment (shyamvalsan) [reply exporting/opentsdb/metadata.yaml:98]: Updated

**Review by sashwathn — DISMISSED** [review]

File: exporting/opentsdb/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+          config: |
+            [opentsdb:my_opentsdb_instance]
+              enabled = yes
+              destination = localhost:4242
+        - name: 'HTTP authentication'
+          folding:
+            enabled: false
+          description: ''
+          config: |
+	     [opentsdb:my_opentsdb_instance]
```
Comment (thiagoftsm) [inline exporting/opentsdb/metadata.yaml:154]: Hello @shyamvalsan ,

`yamllint` reported one more here.

Best regards!

Comment (shyamvalsan) [reply exporting/opentsdb/metadata.yaml]: fixed @thiagoftsm

**Review by thiagoftsm — APPROVED** [review]

**Review by hugovalente-pm — APPROVED** [review]

**Pull request merged**